### PR TITLE
[Feature] Q&A datatype support

### PR DESCRIPTION
**Pull request by rodzy**
## Description of Change(s):

The addition of the [Q&A DataType](https://developers.google.com/search/docs/data-types/qapage) to handle questions and answers with the JSON-LD scripts.

Reference image

Progress:

- [x] Type definitions.
- [x] Structured JSON data.
- [x] Export definitions and props.
- [x] Added example use case.
- [x] Code and test with Cypress.
- [x] Documentation.

---

Some things to help get a PR reviewed and merged faster:

- Have you updated the documentation to go with your changes? **Yes, added all the documentation for all Q&A props** 
- Have you written or updated unit tests? **No**
- Have you written an integration test in the test app supplied? **Yes, I've written e2e tests for the Q&A Page JSON-LD**

### CODE REVIEW

**Comment by rodzy** [comment]
@garmeeh this solves the issue #551, hope this gets merge
Have a great day 😄

File: src/jsonld/qaPage.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    "@context": "https://schema.org",
+    "@type": "QAPage",
+    "mainEntity": ${buildQuestions(mainEntity)}
+    }`;
+  return (
+    <Head>
+      <script
+        type="application/ld+json"
+        dangerouslySetInnerHTML={markup(jslonld)}
+        key="jsonld-qa-page"
```
Comment (garmeeh) [inline src/jsonld/qaPage.tsx:122]: Can you add key override (see other jsonld files for reference)

File: src/jsonld/qaPage.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  upvotedCount?: number;
+  url?: string;
+  author?: Person;
+}
+
+export interface Question {
+  name: string;
+  answerCount: number;
+  acceptedAnswer: Answer;
+  suggestedAnswer: Answer[];
```
Comment (garmeeh) [inline src/jsonld/qaPage.tsx:22]: I believe either or is required 🤔

File: src/jsonld/qaPage.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        }
+        ${
+          mainEntity.author
+            ? `"author": {
+          "@type": "Person",
+          "name": "${mainEntity.author.name}"
+        },`
+            : ''
+        }
+        "acceptedAnswer": {
```
Comment (garmeeh) [inline src/jsonld/qaPage.tsx:56]: As this will become an optional prop we will need to add a check here.

File: src/jsonld/qaPage.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+          ${
+            mainEntity.acceptedAnswer.author
+              ? `"author": {
+            "@type": "Person",
+            "name": "${mainEntity.acceptedAnswer.author.name}"
+          }`
+              : ''
+          }
+        },
+        "suggestedAnswer": [${mainEntity.suggestedAnswer.map(
```
Comment (garmeeh) [inline src/jsonld/qaPage.tsx:83]: As this will become an optional prop we will need to add a check here.

**Review by garmeeh — CHANGES_REQUESTED** [review]
@rodzy thank you for the feature 👏 

A couple of conflicts that need to be sorted and the a few minor change requests. Let me know if you have any questions.

File: src/jsonld/qaPage.tsx (hunk 1 above)
Comment (rodzy) [reply src/jsonld/qaPage.tsx:122]: Understood, working on it

File: src/jsonld/qaPage.tsx (hunk 2 above)
Comment (rodzy) [reply src/jsonld/qaPage.tsx:22]: Yeah, I was thinking on this earlier, I guess I just make them both optional 😁

**Comment by rodzy** [comment]
@garmeeh Everything is resolved for now, thanks for the feedback 👍

**Comment by garmeeh** [comment]
@all-contributors please add @rodzy test

**Review by garmeeh — APPROVED** [review]
💪 Excellent thank you

**Pull request merged**